source/extensions/common/aws/metadata_fetcher.cc

```diff
@@ -23,7 +23,8 @@ class MetadataFetcherImpl : public MetadataFetcher,
   MetadataFetcherImpl(Upstream::ClusterManager& cm, absl::string_view cluster_name)
       : cm_(cm), cluster_name_(std::string(cluster_name)) {}
 
-  ~MetadataFetcherImpl() override { cancel(); }
+  // TODO(suniltheta): Verify that bypassing virtual dispatch here was intentional
```

Comment:
cc @suniltheta

Reply:
LGTM.

Yes, this was intentional. This parent class has multiple sub classes.
Any side effect is doing so? Any thing that will be left not cleaned up properly?

Reply:
>  Any side effect is doing so? Any thing that will be left not cleaned up properly?

@wbpcode ?

Reply:
@suniltheta Calling virtual method in the construct or destructor may result in unexpected error in some cases. And our clang.tidy tools will complain that.
But your code here actually is Okay and this change just want to make the clang.tidy happy.

cc @phlax